flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkDeploymentController.java

```diff
@@ -92,8 +94,11 @@ public DeleteControl cleanup(FlinkDeployment flinkApp, Context context) {
         LOG.info("Stopping cluster {}", flinkApp.getMetadata().getName());
         Configuration effectiveConfig =
                 FlinkUtils.getEffectiveConfig(flinkApp, defaultConfig.getFlinkConfig());
-
-        observer.observe(flinkApp, context, effectiveConfig);
+        try {
+            observer.observe(flinkApp, context, effectiveConfig);
+        } catch (DeploymentFailedException dfe) {
```

Comment:
Does this need to log the exception for the failure of deployment?

Reply:
Relevant information is already logged, for example:
```
[ERROR] [default.basic-example] Deployment default/basic-example failed with pods "basic-example-d5c476cb6-" is forbidden: error looking up service account default/invalid-flink-operator: serviceaccount "invalid-flink-operator" not found
```